plugins/woocommerce/src/Blocks/BlockTypes/FeaturedProduct.php

```diff
@@ -22,7 +22,7 @@ protected function get_item( $attributes ) {
 		$id = absint( $attributes['productId'] ?? 0 );
 
 		$product = wc_get_product( $id );
-		if ( ! $product ) {
+		if ( ! $product || $product->get_status() == 'draft' ) {
```

Comment:
It's not mentioned in the issue, but in addition to `draft`, there are some other statuses which might be good to filter out: `pending`, `private`, `future`...

I wonder if we could do something like `'publish' !== $product->get_status()` instead, so we filter out all products which are not published. What do you think?

Reply:
@Aljullu thanks for the suggestion. I have updated the code.